lib/ansible/cli/config.py

```diff
@@ -0,0 +1,181 @@
+
+        self.set_action()
+
+        # options specific to self.actions
+        if self.action == "list":
+            self.parser.set_usage("usage: %prog list [options] ")
+        if self.action == "dump":
+            self.parser.set_usage("usage: %prog dump [options] [-c ansible.cfg]")
+        elif self.action == "view":
+            self.parser.set_usage("usage: %prog view [options] [-c ansible.cfg] ")
+        elif self.action == "edit":
+            self.parser.set_usage("usage: %prog edit [options] [-c ansible.cfg]")
+        elif self.action == "update":
+            self.parser.add_option('-s', '--setting', dest='setting', help="config setting, the section defaults to 'defaults'")
+            self.parser.set_usage("usage: %prog update [options] [-c ansible.cfg] -s '[section.]setting=value'")
+
+        self.options, self.args = self.parser.parse_args()
+        display.verbosity = self.options.verbosity
+
+    def run(self):
+
+        super(ConfigCLI, self).run()
+
+        if self.options.config_file:
+            self.config_file = unfrackpath(self.options.config_file, follow=False)
+            self.config = ConfigManager(self.config_file)
+        else:
+            self.config = ConfigManager()
+            self.config_file = self.config.data.get_setting('ANSIBLE_CONFIG')
+            try:
+                if not os.path.exists(self.config_file):
+                    raise AnsibleOptionsError("%s does not exist or is not accessible" % (self.config_file))
+                elif not os.path.isfile(self.config_file):
+                    raise AnsibleOptionsError("%s is not a valid file" % (self.config_file))
+
+                os.environ['ANSIBLE_CONFIG'] = self.config_file
+            except:
+                if self.action in ['view']:
+                    raise
+                elif self.action in ['edit', 'update']:
+                    display.warning("File does not exist, used empty file: %s" % self.config_file)
+
+        self.execute()
+
+    def execute_update(self):
+        '''
+        Updates a single setting in the specified ansible.cfg
+        '''
+        raise AnsibleError("Option not implemented yet")
+
+        if self.options.setting is None:
+            raise AnsibleOptionsError("update option requries a setting to update")
+
+        (entry, value) = self.options.setting.split('=')
+        if '.' in entry:
+            (section, option) = entry.split('.')
+        else:
+            section = 'defaults'
+            option = entry
+        subprocess.call([
+            'ansible',
+            '-m','ini_file',
+            'localhost',
+            '-c','local',
+            '-a','"dest=%s section=%s option=%s value=%s backup=yes"' % (self.config_file, section, option, value)
+        ])
+
+    def execute_view(self):
+        '''
+        Displays the current config file
+        '''
+        try:
+            self.pager(open(self.config_file).read())
```

Comment:
This might cause problems with non-ascii but I'm not 100% certain.  On python2, the read() will return a byte string.  On python3, will use the encoding specified via locale (falling back to utf-8 I believe) to decode the bytes into text.

self.pager call display.display() which has a comment that the strings it gets must be text otherwise we'll get tracebacks (That comment seems odd as ansible-doc already calls it with byte strings on python2 I think.  But maybe those are getting coerced on python2 into text or we just don't have any non-ascii chars in the module docs.)